ui/components/component-library/popover/popover.test.tsx

```diff
@@ -0,0 +1,201 @@
+        >
+          {PopoverPosition.Right}
+        </Popover>
+
+        <Popover
+          isOpen={true}
+          data-testid={PopoverPosition.RightStart}
+          position={PopoverPosition.RightStart}
+        >
+          {PopoverPosition.RightStart}
+        </Popover>
+
+        <Popover
+          isOpen={true}
+          data-testid={PopoverPosition.RightEnd}
+          position={PopoverPosition.RightEnd}
+        >
+          {PopoverPosition.RightEnd}
+        </Popover>
+
+        <Popover
+          isOpen={true}
+          data-testid={PopoverPosition.Bottom}
+          position={PopoverPosition.Bottom}
+        >
+          {PopoverPosition.Bottom}
+        </Popover>
+
+        <Popover
+          isOpen={true}
+          data-testid={PopoverPosition.BottomStart}
+          position={PopoverPosition.BottomStart}
+        >
+          {PopoverPosition.BottomStart}
+        </Popover>
+
+        <Popover
+          isOpen={true}
+          data-testid={PopoverPosition.BottomEnd}
+          position={PopoverPosition.BottomEnd}
+        >
+          {PopoverPosition.BottomEnd}
+        </Popover>
+
+        <Popover
+          isOpen={true}
+          data-testid={PopoverPosition.Left}
+          position={PopoverPosition.Left}
+        >
+          {PopoverPosition.Left}
+        </Popover>
+
+        <Popover
+          isOpen={true}
+          data-testid={PopoverPosition.LeftStart}
+          position={PopoverPosition.LeftStart}
+        >
+          {PopoverPosition.LeftStart}
+        </Popover>
+
+        <Popover
+          isOpen={true}
+          data-testid={PopoverPosition.LeftEnd}
+          position={PopoverPosition.LeftEnd}
+        >
+          {PopoverPosition.LeftEnd}
+        </Popover>
+      </>,
+    );
+
+    expect(getByText(PopoverPosition.Auto)).toBeDefined();
+    expect(getByText(PopoverPosition.Top)).toBeDefined();
+    expect(getByText(PopoverPosition.TopStart)).toBeDefined();
+    expect(getByText(PopoverPosition.TopEnd)).toBeDefined();
+    expect(getByText(PopoverPosition.Right)).toBeDefined();
+    expect(getByText(PopoverPosition.RightStart)).toBeDefined();
+    expect(getByText(PopoverPosition.RightEnd)).toBeDefined();
+    expect(getByText(PopoverPosition.Bottom)).toBeDefined();
+    expect(getByText(PopoverPosition.BottomStart)).toBeDefined();
+    expect(getByText(PopoverPosition.BottomEnd)).toBeDefined();
+    expect(getByText(PopoverPosition.Left)).toBeDefined();
+    expect(getByText(PopoverPosition.LeftStart)).toBeDefined();
+    expect(getByText(PopoverPosition.LeftEnd)).toBeDefined();
+  });
+
+  it('should render an arrow on popover', () => {
+    const { getByTestId } = render(
+      <Popover data-testid="popover" isOpen={true} hasArrow>
+        Popover
+      </Popover>,
+    );
+
+    const arrowElement =
+      getByTestId('popover').querySelector('.mm-popover__arrow');
+    expect(arrowElement).toHaveClass('mm-popover__arrow');
+  });
+});
```

Comment:
Can we improve some of the testing coverage here by testing some more of the props not listed here?